src/main/kotlin/com/openlattice/shuttle/controllers/ShuttleController.kt

```diff
@@ -0,0 +1,103 @@
+
+    @Timed
+    @GetMapping(path = [INTEGRATION_NAME_PATH + INTEGRATION_KEY_PATH + CALLBACK_PATH])
+    override fun enqueueIntegration(
+            @PathVariable(INTEGRATION_NAME) integrationName: String,
+            @PathVariable(INTEGRATION_KEY) integrationKey: UUID,
+            @PathVariable(CALLBACK) callbackUrl: Optional<String>
+    ): UUID {
+        val normalizedName = normalizeIntegrationName(integrationName)
+        return integrationService.enqueueIntegrationJob(normalizedName, integrationKey, callbackUrl)
+    }
+
+    @Timed
+    @GetMapping(path = [STATUS_PATH + JOB_ID_PATH])
+    override fun pollIntegration(
+            @PathVariable(JOB_ID) jobId: UUID
+    ): IntegrationStatus {
+        ensureAdminAccess()
+        return integrationService.pollIntegrationStatus(jobId)
+    }
+
+    @Timed
+    @GetMapping(path = [STATUS_PATH])
+    override fun pollAllIntegrations(): Map<UUID, IntegrationJob> {
+        ensureAdminAccess()
+        return integrationService.pollAllIntegrationStatuses()
+    }
+
+    @Timed
+    @PostMapping(path = [DEFINITION_PATH + INTEGRATION_NAME_PATH])
+    override fun createIntegrationDefinition(
+            @PathVariable integrationName: String,
+            @RequestBody integrationDefinition: Integration): UUID {
+        ensureAdminAccess()
+        val normalizedName = normalizeIntegrationName(integrationName)
+        return integrationService.createIntegrationDefinition(normalizedName, integrationDefinition)
+    }
+
+    @Timed
+    @GetMapping(path = [DEFINITION_PATH + INTEGRATION_NAME_PATH])
+    override fun readIntegrationDefinition(
+            @PathVariable integrationName: String ) : Integration {
+        ensureAdminAccess()
+        val normalizedName = normalizeIntegrationName(integrationName)
+        return integrationService.readIntegrationDefinition(normalizedName)
+    }
+
+    @Timed
+    @PatchMapping(path = [DEFINITION_PATH + INTEGRATION_NAME_PATH])
+    override fun updateIntegrationDefinition(
+            @PathVariable integrationName: String,
+            @RequestBody integrationUpdate: IntegrationUpdate) {
+        ensureAdminAccess()
+        val normalizedName = normalizeIntegrationName(integrationName)
+        integrationService.updateIntegrationDefinition(normalizedName, integrationUpdate)
+    }
+
+    @Timed
+    @DeleteMapping(path = [DEFINITION_PATH + INTEGRATION_NAME_PATH])
+    override fun deleteIntegrationDefinition(
+            @PathVariable integrationName: String
+    ) {
+        ensureAdminAccess()
+        val normalizedName = normalizeIntegrationName(integrationName)
+        integrationService.deleteIntegrationDefinition(normalizedName)
+    }
+
+    override fun getAuthorizationManager(): AuthorizationManager {
+        return authorizationManager
+    }
+
+    private fun normalizeIntegrationName(integrationName: String): String {
+        return integrationName.toLowerCase().trim()
```

Comment:
This might be the micro-est of optimizations ever but `.trim()` then `.toLowerCase()` :face_with_monocle: